custom_components/google_home/sensor.py

```diff
@@ -40,17 +40,21 @@ async def async_setup_entry(
                 device.name,
             )
         )
-        if device.auth_token and device.available:
-            sensors += [
-                GoogleHomeAlarmsSensor(
-                    coordinator,
-                    device.name,
-                ),
-                GoogleHomeTimersSensor(
-                    coordinator,
-                    device.name,
-                ),
-            ]
+        if device.auth_token:
+            if device.has_alarms():
```

Comment:
This looks like that sensor won't be created if device doesn't have any alarms *set*.
What will be `self._alarms` value when device supports alarms but they're not set?

Reply:
Yeh, that what i meant in descriptions:

> The behavior is not ideal still. In sensor.py under async_setup_entry(once during HA bootup) we setup sensors if the device has token and timers/alarms, but what happens to devices that are not yet available(like offline), those will be forever skipped, since no sensors will be created for those devices until HA is restarted again. If that's true - it defeats the purpose of #122

`self._alarms` is just an empty list if no alarms were fetched during api call, right ?